tests/substrate_tests/calls.rs

```diff
@@ -216,6 +216,32 @@ fn try_catch_external_calls() {
     runtime.constructor(0, Vec::new());
     runtime.function_expect_failure("test", Vec::new());
 
+    let mut runtime = build_solidity(
+        r##"
+        contract c {
+            function test() public {
+                other o = new other();
+                int32 x = 0;
+                try o.test() {
+                    x = 1;
+                } catch (bytes c) {
+                    assert(c == hex"a079c3080c666f6f");
+                    x = 2;
+                }
+                assert(x == 2);
```

Comment:
Without further context, I couldn't tell why this test case exists or what exactly is being tested here.

I think it'd be better if this `test()` function would return `x` instead, so that it can be explicitly checked later on. Also it could go into a dedicated `#[test]`.

Reply:
I've moved the test into it's own `#[test]` and added some comments, and expanded the description on this PR.

Reply:
Much clearer now, thanks!